sklearn/decomposition/_online_lda_fast.pyx

```diff
@@ -66,7 +70,7 @@ def _dirichlet_expectation_2d(cnp.ndarray[ndim=2, dtype=floating] arr):
     the exp and doesn't add in the prior.
     """
     cdef floating row_total, psi_row_total
-    cdef cnp.ndarray[ndim=2, dtype=floating] d_exp
+    cdef floating[:, :] d_exp
```

Comment:
You have a missing const-qualification here.

Reply:
This is being modified ahead
https://github.com/scikit-learn/scikit-learn/pull/25020/files#diff-8b143ebc64c5276c50b2100f5686261cf87c050028876df9b84c056fd3194bacR87